### PR TITLE
One-way communication offline funds transfer using QR-code

### PR DESCRIPTION
**Pull request by mitchellolsthoorn**
This PR adds the ability to export funds without a network connection by using a throw-away identity for transferring between the two different identities

### CODE REVIEW

**Comment by mitchellolsthoorn** [comment]
@devos50 Can you review the PR for merge?

**Comment by devos50** [comment]
@mitchellolsthoorn please prefix the PR title with `READY` instead of `REVIEW` ;)

**Comment by devos50** [comment]
#3364 fixes the unit tests. I would like to merge that one first. After that's done, please rebase this PR and I will review it.

**Comment by devos50** [comment]
@mitchellolsthoorn please rebase this PR on devel again. The tests should pass now.

**Comment by mitchellolsthoorn** [comment]
@devos50 I was planning on redoing the GUI changes on top of the latest commit like you suggested but I discovered that the latest commit also has the bug, so the bug is not introduced by the rebase.

screenshot from 2018-01-22 13-09-07

All the settings tabs are shifted and some options are missing. Do you want me to rebase it on the broken version, or wait till its fixed?

File: Tribler/Core/Modules/restapi/trustchain_endpoint.py [hunk 1]
```diff
@@ -13,7 +13,10 @@ class TrustchainEndpoint(resource.Resource):
     def __init__(self, session):
         resource.Resource.__init__(self)
 
-        child_handler_dict = {"statistics": TrustchainStatsEndpoint, "blocks": TrustchainBlocksEndpoint}
+        child_handler_dict = {
+            "statistics": TrustchainStatsEndpoint,
+            "blocks": TrustchainBlocksEndpoint,
+            "bootstrap": TrustchainBootstrapEndpoint}
```
Comment (devos50) [inline Tribler/Core/Modules/restapi/trustchain_endpoint.py]: Please position the `}` on a new line.

File: Tribler/Core/Modules/restapi/trustchain_endpoint.py [hunk 2]
```diff
@@ -161,3 +164,69 @@ def render_GET(self, request):
...
+            return json.dumps({"error": "trustchain community not found"})
+
+        statistics = mc_community.get_statistics()
+        up = statistics['total_up']
+        down = statistics['total_down']
+        if 'up' in request.args:
+            try:
+                up = int(request.args['up'][0])
+            except ValueError:
+                up = up
```
Comment (devos50) [inline Tribler/Core/Modules/restapi/trustchain_endpoint.py]: Please raise an error here when the input is not a valid integer so it's consistent with the code in other endpoints.

File: Tribler/community/triblerchain/community.py [hunk 3]
```diff
@@ -215,6 +215,40 @@ def get_trust(self, member):
             # We need a minimum of 1 trust to have a chance to be selected in the categorical distribution.
             return 1
 
+    def bootstrap_new_identity(self, up, down):
+        """ 
+        One-way payment channel.
+        Create a new temporary identity , and transfer funds to the new identity.
```
Comment (devos50) [inline Tribler/community/triblerchain/community.py]: Remove the whitespace before the comma.

File: Tribler/community/triblerchain/community.py [hunk 4]
```diff
@@ -232,3 +266,8 @@ def on_introduction_response(self, messages):
 
     def start_walking(self):
         self.register_task("take step", LoopingCall(self.take_step)).start(self.CrawlerDelay, now=False)
+
+
+class DBShim:
```
Comment (devos50) [inline Tribler/community/triblerchain/community.py]: Please inherit this class from `object`. Also, clarify why you need this class.

File: tribler-headless.sh [hunk 5]
```diff
@@ -0,0 +1 @@
+python2 -m twisted tribler --restapi 8080 --manhole 3000
```
Comment (devos50) [inline tribler-headless.sh]: Please remove this file. We already have a `--headless` option.

**Comment by devos50** [comment]
@mitchellolsthoorn please rebase onto the latest version of `devel` (sorry for the merge conflict, this should be the final time to rebase). All tests should pass now.

Also, I thought a bit about the GUI test for the QR code. If you want to make this, you also need to adjust the fake Tribler API (https://github.com/Tribler/fake-tribler-api) and I don't think it's worth the time to dive into this so I'm fine with it if you don't create a GUI test. I would like to see a test for the API endpoints, however. Please refer to the other REST API tests to get an idea of how to write them.

If you have any other questions, please let me know 👍

**Comment by mitchellolsthoorn** [comment]
@devos50 I rebased it on devel, covered all the point you had and wrote some tests

**Comment by devos50** [comment]
Looks good, thanks!

**Pull request merged**